plugins/woocommerce-blocks/assets/js/blocks-registry/product-collection/register-product-collection.tsx

```diff
@@ -26,17 +27,364 @@ export interface ProductCollectionConfig extends BlockVariation {
+		return false;
+	}
+	if (
+		config.isActive !== undefined &&
+		typeof config.isActive !== 'function'
+	) {
+		console.error( 'Invalid isActive: isActive must be a function.' );
+		return false;
+	}
+
+	/**
+	 * Attributes validation
+	 */
+	if (
+		config.attributes !== undefined &&
+		typeof config.attributes !== 'object'
+	) {
+		console.error( 'Invalid attributes: attributes must be an object.' );
+		return false;
+	}
+	if (
+		config.attributes?.query !== undefined &&
+		typeof config.attributes.query !== 'object'
+	) {
+		console.error( 'Invalid query: query must be an object.' );
+		return false;
+	}
+	//attributes. query.exclude
+	if (
+		config.attributes?.query?.exclude !== undefined &&
+		! Array.isArray( config.attributes.query.exclude )
+	) {
+		console.error( 'Invalid exclude: exclude must be an array.' );
+		return false;
+	}
+	// attributes.query.offset
+	if (
+		config.attributes?.query?.offset !== undefined &&
+		typeof config.attributes.query.offset !== 'number'
+	) {
+		console.error( 'Invalid offset: offset must be a number.' );
+		return false;
+	}
+	// attributes.query.order
+	if (
+		config.attributes?.query?.order !== undefined &&
+		typeof config.attributes.query.order !== 'string'
+	) {
+		console.error( 'Invalid order: order must be a string.' );
+		return false;
+	}
+	// attributes.query.orderBy
+	if (
+		config.attributes?.query?.orderBy !== undefined &&
+		typeof config.attributes.query.orderBy !== 'string'
+	) {
+		console.error( 'Invalid orderBy: orderBy must be a string.' );
+		return false;
+	}
+	// attributes.query.pages
+	if (
+		config.attributes?.query?.pages !== undefined &&
+		typeof config.attributes.query.pages !== 'number'
+	) {
+		console.error( 'Invalid pages: pages must be a number.' );
+		return false;
+	}
+	// attributes.query.perPage
+	if (
+		config.attributes?.query?.perPage !== undefined &&
+		typeof config.attributes.query.perPage !== 'number'
+	) {
+		console.error( 'Invalid perPage: perPage must be a number.' );
+		return false;
+	}
+	// attributes.query.search
+	if (
+		config.attributes?.query?.search !== undefined &&
+		typeof config.attributes.query.search !== 'string'
+	) {
+		console.error( 'Invalid search: search must be a string.' );
+		return false;
+	}
+	// attributes.query.taxQuery
+	if (
+		config.attributes?.query?.taxQuery !== undefined &&
+		typeof config.attributes.query.taxQuery !== 'object'
+	) {
+		console.error( 'Invalid taxQuery: taxQuery must be an object.' );
+		return false;
+	}
+	// attributes.query.featured
+	if (
+		config.attributes?.query?.featured !== undefined &&
+		typeof config.attributes.query.featured !== 'boolean'
+	) {
+		console.error( 'Invalid featured: featured must be a boolean.' );
+		return false;
+	}
+	// attributes.query.timeFrame
+	if (
+		config.attributes?.query?.timeFrame !== undefined &&
+		typeof config.attributes.query.timeFrame !== 'object'
+	) {
+		console.error( 'Invalid timeFrame: timeFrame must be an object.' );
+		return false;
+	}
```

Comment:
**Question:** How deep should we go for validation? For example, when I am validating `attributes.query.timeFrame`. Should I validate `operator` and `value` properties of `TimeFrame` object?

Here are some more such examples:
- woocommerceStockStatus: Should I validate if each array item is a valid value?
- query.priceRange: Should I validate if `min` and `max` are numbers?
- displayLayout: Should I validate if `type` is a valid value from `LayoutOptions` enum? Should I validate if `columns` is a number and `shrinkColumns` is a boolean?
- query.order: Should I validate if it's a valid value i.e. 'asc' or 'desc'?
- query.orderBy: Should I validate if it's a valid value i.e. 'date', 'title', 'popularity', 'rating'?

There could be other such properties too. I just wanna understand sentiment on how deep should we go for validation. Based on that I can update the PR. Looking forward to your thoughts.

Reply:
I don't think that's necessary.

Reply:
Sounds good! Let's keep validation top-level only. If needed, we will make it more strict in future.  CC: @xristos3490